include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -108,92 +115,184 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
         }
 
+        /** @brief Returns the coordinate of the first support knot associated to a discrete_element identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_first_support_knot(
                 discrete_element_type const& ix) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid()));
         }
 
+        /** @brief Returns the coordinate of the last support knot associated to a discrete_element identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_last_support_knot(
                 discrete_element_type const& ix) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid() + degree() + 1));
         }
 
+        /** @brief Returns the coordinate of the \f$n\f$-th knot associated to a DiscreteElement identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @param[in] n Integer indexing a knot in the support of the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_support_knot_n(
                 discrete_element_type const& ix,
                 int n) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid() + n));
         }
 
+        /** @brief Returns the coordinate of the lower boundary knot.
+         *
+         * @return Coordinate of the lower boundary knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> rmin() const noexcept
         {
             return get_knot(0);
         }
 
+        /** @brief Returns the coordinate of the upper boundary knot.
+         *
+         * @return Coordinate of the upper boundary knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> rmax() const noexcept
         {
             return get_knot(ncells());
         }
 
+        /** @brief Returns the length of the domain supporting knots.
+         *
+         * @return The length of the domain supporting knots.
+         */
         KOKKOS_INLINE_FUNCTION double length() const noexcept
         {
             return rmax() - rmin();
         }
 
+        /** @brief TODO
+         *
+         * @return TODO.
+         */
         KOKKOS_INLINE_FUNCTION std::size_t size() const noexcept
         {
             return degree() + ncells();
         }
 
-        /// Returns the discrete domain including ghost bsplines
+        /** @brief Returns the discrete domain including ghost bsplines
+         *
+         * @return The discrete domain including ghost bsplines.
+         */
         KOKKOS_INLINE_FUNCTION discrete_domain_type full_domain() const
```

Comment:
Are ghost b-splines what I am describing in the docstring above? If so then one or both of these docstrings should define that term